genie-agent/src/main/java/com/netflix/genie/agent/execution/process/impl/JobProcessManagerImpl.java

```diff
@@ -195,11 +198,14 @@ public JobStatus waitFor() throws InterruptedException {
         }
 
         if (killed.get()) {
-            return JobStatus.KILLED;
-        } else if (exitCode == 0) {
-            return JobStatus.SUCCEEDED;
+            return new JobProcessResult.Builder(JobStatus.KILLED, JobStatusMessages.JOB_KILLED_BY_USER).build();
+        } else if (exitCode == SUCCESS_EXIT_CODE) {
+            return new JobProcessResult.Builder(
+                JobStatus.SUCCEEDED,
+                JobStatusMessages.JOB_FINISHED_SUCCESSFULLY
+            ).build();
         } else {
-            return JobStatus.FAILED;
+            return new JobProcessResult.Builder(JobStatus.FAILED, JobStatusMessages.JOB_FAILED).build();
```

Comment:
Seems like this loses the exit code. I'm guessing you probably want to save it.
(I would also put it in the final status message, users do look at it and special codes have meaning that people need to know).

Reply:
Yeah we probably want exit code in the return that's true. I'll add it in a subsequent commit